File: Python/Program/Mahasiswa.py
# Saya Farhan Muzhaffar Tiras Putra NIM 2105879 mengerjakan soal Latihan 2 dalam mata kuliah Desain dan Pemrograman Berorientasi Objek
# untuk keberkahanNya maka saya tidak melakukan kecurangan seperti yang telah dispesifikasikan.
# Aamiin.

from SivitasAkademik import SivitasAkademik

# deklarasi class

# deklarasi class menjadi anak dari SivitasAkademik


class Mahasiswa(SivitasAkademik):

    # deklarasi private atribut
    __nim = ""
    __nama = ""
    __prodi = ""
    __fakultas = ""

    # constructor dengan parameter nim, nama, prodi, dan fakultas
    def __init__(self, nim="", nama="", prodi="", fakultas=""):
        self.__nim = nim
        self.__nama = nama
        self.__prodi = prodi
        self.__fakultas = fakultas

    # setter dan getter untuk masing masing attribut nim, nama, prodi dan fakultas
    def get_nim(self):
        return self.__nim

    def set_nim(self, nim):
        self.__nim = nim

    def get_nama(self):
        return self.__nama

    def set_nama(self, nama):
        self.__nama = nama

    def get_prodi(self):
        return self.__prodi

    def set_prodi(self, prodi):
        self.__prodi = prodi

    def get_fakultas(self):
        return self.__fakultas

    def set_fakultas(self, fakultas):
        self.__fakultas = fakultas

File: PHP/Program/Mahasiswa.php
<?php
/* Saya Farhan Muzhaffar Tiras Putra NIM 2105879 mengerjakan soal Latihan 2 dalam mata kuliah Desain dan Pemrograman Berorientasi Objek
untuk keberkahanNya maka saya tidak melakukan kecurangan seperti yang telah dispesifikasikan.
Aamiin. */

require('SivitasAkademik.php');

// deklarasi class menjadi anak dari SivitasAkademik

class Mahasiswa extends SivitasAkademik
{
    // attibut private
    private $nim;
    private $nama;
    private $prodi;
    private $fakultas;

    // constructor dengan parameter nim, nama, prodi, dan fakultas
    public function __construct($nim = '', $nama = '', $prodi = '', $fakultas = '')
    {
        $this->nim = $nim;
        $this->nama = $nama;
        $this->prodi = $prodi;
        $this->fakultas = $fakultas;
    }

    // setter dan getter untuk setiap atribut nim, nama, prodi dan fakultas
    public function get_nim()
    {
        return $this->nim;
    }
    public function set_nim($nim)
    {
        $this->nim = $nim;
    }

    public function get_nama()
    {
        return $this->nama;
    }
    public function set_nama($nama)
    {
        $this->nama = $nama;
    }

    public function get_prodi()
    {
        return $this->prodi;
    }
    public function set_prodi($prodi)
    {
        $this->prodi = $prodi;
    }

    public function get_fakultas()
    {
        return $this->fakultas;
    }
    public function set_fakultas($fakultas)
    {
        $this->fakultas = $fakultas;
    }
};
